### PR TITLE
Bd 1.0.0

### PR DESCRIPTION
**Pull request by BDonnot**
Improving the speed of "get_topological_impact" to reduce the computation time
Fix a bug in the "connexity converter"

### CODE REVIEW

**Review by BDonnot — COMMENTED** [review]
Files changed look good

File: grid2op/Action/BaseAction.py [hunk 1]
```diff
@@ -462,87 +459,32 @@ def get_topological_impact(self, powerline_status=None):
...
-                beg_ += nb_obj
+            isnotconnected = np.full(self.n_line, fill_value=True, dtype=dt_bool)
+        else:
+            isnotconnected = ~powerline_status
+
+        self._lines_impacted = self._switch_line_status | (self._set_line_status != 0)
+        self._subs_impacted = np.full(shape=self.sub_info.shape, fill_value=False, dtype=dt_bool)
+
+        # todo could be set as a class attribute
+        _topo_vect_to_sub = np.repeat(np.arange(self.sub_info.shape[0]), repeats=self.sub_info)
```
Comment (Tezirg) [inline grid2op/Action/BaseAction.py]: `self.sub_info.shape[0] == self.n_sub`

File: grid2op/Action/BaseAction.py [hunk 2]
```diff
@@ -462,87 +459,32 @@ def get_topological_impact(self, powerline_status=None):
...
         on the powergrid.
-
-
         Returns
         -------
         lines_impacted: :class:`numpy.array`, dtype:dt_bool
             A vector with the same size as the number of powerlines in the grid (:attr:`BaseAction.n_line`) with for each
             component ``True`` if the line STATUS is impacted by the action, and ``False`` otherwise. See
             :attr:`BaseAction._lines_impacted` for more information.
-
```
Comment (Tezirg) [inline grid2op/Action/BaseAction.py]: The methods still returns subs_impacted

**Review by Tezirg — APPROVED** [review]
Looks efficient and well tested, :noice:

**Review by Tezirg — APPROVED** [review]

**Pull request merged**